opm/simulators/wells/BlackoilWellModelGeneric.cpp

```diff
@@ -459,24 +457,32 @@ checkGroupInjectionConstraints(const Group& group,
             }
         }
     }
-    if (controls.has_control(Group::InjectionCMode::RESV))
+    if (group.has_control(phase, Group::InjectionCMode::RESV))
     {
         if (currentControl != Group::InjectionCMode::RESV)
         {
             double current_rate = 0.0;
             current_rate += WellGroupHelpers::sumWellResRates(group, schedule(), well_state, reportStepIdx, phasePos, /*isInjector*/true);
             // sum over all nodes
             current_rate = comm_.sum(current_rate);
+            const auto& controls = group.injectionControls(phase);
+            auto target = controls.resv_max_rate;
+
+            if (group.has_gpmaint_control(phase, currentControl)) {
+                const auto& gpmaint = group.gpmaint().value();
+                auto& gpmaint_state = group_state.gpmaint();
+                target = gpmaint.rate(gpmain_state, current_rate, error, dt);
+            }
```

Comment:
This is not correct, you should get the rate from the group_state directly. The update happens elsewhere. I suggest you remove the gpmaint part of this and I will add it to my PR. Then it can be merged safely together with your opm-common PR.

Reply:
Ok cool - Will do (tomorrow?)